src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java

```diff
@@ -63,8 +64,7 @@ public QueryResponseBean(QueryScreen queryScreen, SessionWrapper session) {
                     }
                 }
                 if (indicesForSelectedChoices.size() > 0) {
-                    currentAnswer = String.join(RemoteQuerySessionManager.ANSWER_DELIMITER,
-                            indicesForSelectedChoices);
+                    currentAnswer = DataUtil.joinWithSpaces(indicesForSelectedChoices.toArray(new String[0]));
```

Comment:
we should initialise the array with the exact size like - 
```suggestion
                    currentAnswer = DataUtil.joinWithSpaces(indicesForSelectedChoices.toArray(new String[indicesForSelectedChoices.size()]));
```

Reply:
b744e383